### PR TITLE
ppl: add checks to ensure properly placement of pin groups

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

**Comment by eder-matheus** [comment]
@maliberty My latest commit closes the more significant part of the pin grouping approach inside the SA. This is ready for review.

After getting this merged, I'll go to the io constraints.

File: src/ppl/test/annealing2.ok [hunk 1]
```diff
@@ -10,5 +10,5 @@
 Found 0 macro blocks.
 Using 1u default distance from corners.
 Using 2 tracks default min distance between IO pins.
-[INFO PPL-0012] I/O nets HPWL: 1962.48 um.
+[INFO PPL-0012] I/O nets HPWL: 1962.64 um.
```
Comment (maliberty) [inline src/ppl/test/annealing2.ok:13]: Are you expecting a change in this test which doesn't use groups?

**Comment by maliberty** [comment]
I think we are going to need more way to move/swap groups.  This set of choices seems to make a hard separation between lone and group pins but we may need to be able to operate across that barrier.

File: src/ppl/test/annealing2.ok (hunk 1 above)
Comment (eder-matheus) [reply src/ppl/test/annealing2.ok:13]: This test has two pin groups.

**Comment by eder-matheus** [comment]
> I think we are going to need more way to move/swap groups. This set of choices seems to make a hard separation between lone and group pins but we may need to be able to operate across that barrier.

As mentioned in the other PR, I agree with that. After having the PR https://github.com/The-OpenROAD-Project/OpenROAD/pull/3603 and this PR merged, I'll work on allowing the swap between pin groups and lone pins.

**Review by maliberty — APPROVED** [review]

**Pull request merged**